fury/ui/tests/test_containers.py

```diff
@@ -360,3 +361,117 @@ def tab_change(tab_ui):
+    tab_ui.tabs[1].title_color = (0, 1, 0)
+    tab_ui.tabs[2].title_color = (0, 0, 1)
+
+    npt.assert_equal(tab_ui.tabs[0].title_color, (1., .0, .0))
+    npt.assert_equal(tab_ui.tabs[1].title_color, (.0, 1., .0))
+    npt.assert_equal(tab_ui.tabs[2].title_color, (.0, .0, 1.))
+
+    npt.assert_equal(tab_ui.tabs[0].title_font_size, 18)
+    npt.assert_equal(tab_ui.tabs[1].title_font_size, 18)
+    npt.assert_equal(tab_ui.tabs[2].title_font_size, 18)
+
+    tab_ui.tabs[0].title_font_size = 10
+    tab_ui.tabs[1].title_font_size = 20
+    tab_ui.tabs[2].title_font_size = 30
+
+    npt.assert_equal(tab_ui.tabs[0].title_font_size, 10)
+    npt.assert_equal(tab_ui.tabs[1].title_font_size, 20)
+    npt.assert_equal(tab_ui.tabs[2].title_font_size, 30)
+
+    npt.assert_equal(tab_ui.tabs[0].title_italic, False)
+    npt.assert_equal(tab_ui.tabs[1].title_italic, False)
+    npt.assert_equal(tab_ui.tabs[2].title_italic, False)
+
+    tab_ui.tabs[0].title_italic = False
+    tab_ui.tabs[1].title_italic = True
+    tab_ui.tabs[2].title_italic = False
+
+    npt.assert_equal(tab_ui.tabs[0].title_italic, False)
+    npt.assert_equal(tab_ui.tabs[1].title_italic, True)
+    npt.assert_equal(tab_ui.tabs[2].title_italic, False)
+
+    tab_ui.add_element(0, ui.Checkbox(['Option 1', 'Option 2']), (0.5, 0.5))
+    tab_ui.add_element(1, ui.LineSlider2D(), (0.0, 0.5))
+    tab_ui.add_element(2, ui.TextBlock2D(), (0.5, 0.5))
+
+    with npt.assert_raises(IndexError):
+        tab_ui.add_element(3, ui.TextBlock2D(), (0.5, 0.5, 0.5))
+
+    with npt.assert_raises(IndexError):
+        tab_ui.remove_element(3, ui.TextBlock2D())
+
+    with npt.assert_raises(IndexError):
+        tab_ui.update_element(3, ui.TextBlock2D(), (0.5, 0.5, 0.5))
+
+    npt.assert_equal('Tab 1', tab_ui.tabs[0].title)
+    npt.assert_equal('Tab 2', tab_ui.tabs[1].title)
+    npt.assert_equal('Tab 3', tab_ui.tabs[2].title)
+
+    npt.assert_equal(3, tab_ui.nb_tabs)
+
+    collapses = itertools.count()
+    changes = itertools.count()
+
+    def collapse(tab_ui):
+        if tab_ui.collapsed:
+            next(collapses)
+
+    def tab_change(tab_ui):
+        next(changes)
+
+    tab_ui.on_change = tab_change
+    tab_ui.on_collapse = collapse
+
+    event_counter = EventCounter()
+    event_counter.monitor(tab_ui)
+
+    current_size = (800, 800)
+    show_manager = window.ShowManager(size=current_size, title='Tab UI Test')
+    show_manager.scene.add(tab_ui)
+
+    if interactive:
+        show_manager.record_events_to_file(recording_filename)
+        print(list(event_counter.events_counts.items()))
+        event_counter.save(expected_events_counts_filename)
+    else:
+        show_manager.play_events_from_file(recording_filename)
+        expected = EventCounter.load(expected_events_counts_filename)
+        event_counter.check_counts(expected)
+
+    npt.assert_equal(0, tab_ui.active_tab_idx)
+    npt.assert_equal(11, next(changes))
+    npt.assert_equal(4, next(collapses))
```

Comment:
Although I understand your motivation, try to include both the `top` and `bottom` tests in the `test_ui_tab_ui`. If you need a specific test for each, then please make sure to make it specific, since this solution duplicates code and complicates maintenance in the long term.

Reply:
sure will try to do make a combined test for both